internal/sdkprovider/service/kafka/kafka_quota.go

```diff
@@ -0,0 +1,217 @@
+	)
+
+	if err := schemautil.ResourceDataGet(
+		d,
+		&req,
+		schemautil.RenameAlias("client_id", "client-id"),
+	); err != nil {
+		return err
+	}
+
+	if err := client.ServiceKafkaQuotaCreate(ctx, project, service, &req); err != nil {
+		return err
+	}
+
+	d.SetId(schemautil.BuildResourceID(project, service, clientID, user))
+
+	return resourceKafkaQuotaRead(ctx, d, client)
+}
+
+func resourceKafkaQuotaUpdate(ctx context.Context, d *schema.ResourceData, client avngen.Client) error {
+	var (
+		req kafka.ServiceKafkaQuotaCreateIn
+	)
+
+	project, service, _, _, err := schemautil.SplitResourceID4(d.Id())
+	if err != nil {
+		return err
+	}
+
+	if err := schemautil.ResourceDataGet(
+		d,
+		&req,
+		schemautil.RenameAlias("client_id", "client-id"),
+	); err != nil {
+		return err
+	}
+
+	if err := client.ServiceKafkaQuotaCreate(ctx, project, service, &req); err != nil {
+		return err
+	}
+
+	return resourceKafkaQuotaRead(ctx, d, client)
+}
+
+func resourceKafkaQuotaRead(ctx context.Context, d *schema.ResourceData, client avngen.Client) error {
+	project, serviceName, clientID, user, err := schemautil.SplitResourceID4(d.Id())
+	if err != nil {
+		return err
+	}
+
+	var params [][2]string
+	if user != "" {
+		params = append(params, kafka.ServiceKafkaQuotaDescribeUser(user))
+	}
+
+	if clientID != "" {
+		params = append(params, kafka.ServiceKafkaQuotaDescribeClientId(clientID))
+	}
+
+	resp, err := client.ServiceKafkaQuotaDescribe(
+		ctx,
+		project,
+		serviceName,
+		params...,
+	)
+	if err != nil {
+		return err
+	}
+
+	return schemautil.ResourceDataSet(
+		aivenKafkaQuotaSchema,
+		d,
+		resp,
+		schemautil.RenameAlias("client_id", "client-id"),
+	)
+}
+
+func resourceKafkaQuotaDelete(ctx context.Context, d *schema.ResourceData, client avngen.Client) error {
+	var (
+		project     = d.Get("project").(string)
+		serviceName = d.Get("service_name").(string)
+		clientID    = d.Get("client_id").(string)
+		user        = d.Get("user").(string)
+	)
+
+	var params [][2]string
+	if user != "" {
+		params = append(params, kafka.ServiceKafkaQuotaDeleteUser(user))
+	}
+
+	if clientID != "" {
+		params = append(params, kafka.ServiceKafkaQuotaDeleteClientId(clientID))
+	}
+
+	return client.ServiceKafkaQuotaDelete(
+		ctx,
+		project,
+		serviceName,
+		params...,
+	)
+}
+
+func validateKafkaQuotaDiff(_ context.Context, d *schema.ResourceDiff, _ interface{}) error {
+	var (
+		user     = d.Get("user").(string)
+		clientID = d.Get("client_id").(string)
+	)
+
+	if user == "" && clientID == "" {
+		return fmt.Errorf("at least one of user or client_id must be specified")
+	}
+
+	var (
+		consumerByteRate  = d.Get("consumer_byte_rate").(int)
+		producerByteRate  = d.Get("producer_byte_rate").(int)
+		requestPercentage = d.Get("request_percentage").(int)
+	)
+
+	if consumerByteRate == 0 && producerByteRate == 0 && requestPercentage == 0 {
+		return fmt.Errorf("at least one quota parameter must be specified")
+	}
+
+	return nil
```

Comment:
1. There is an attribute `AtLeastOneOf`, probably can be used with client id and the user https://developer.hashicorp.com/terraform/plugin/framework/migrating/attributes-blocks/validators-predefined#sdkv2
2. Zero value can't be set, right?